### PR TITLE
(kitty) Fix update.ps1 and chocolateyInstall.ps1 scripts for Kitty package

### PR DESCRIPTION
**Pull request by yelkarama**
## Description
Fix update.ps1 to correctly fetch asset (zip file) from github (using GitHub REST api) and modify chocolateyInstall.ps1 to unzip the downloaded zip file

## Motivation and Context
kitty was not able to upgrade to new available versions

## How Has this Been Tested?
Used chocolatey test environment. Tested new install and force reinstall scenarios.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Migrated package (a package has been migrated from another repository)

## Checklist:

- [x] My code follows the code style of this repository.
- [ ] My change requires a change to documentation (this usually means the notes in the description of a package).
- [ ] I have updated the documentation accordingly (this usually means the notes in the description of a package).
- [ ] I have updated the package description and it is less than 4000 characters.
- [x] All files are up to date with the latest [Contributing Guidelines](https://github.com/chocolatey-community/chocolatey-packages/blob/master/CONTRIBUTING.md)
- [x] The added/modified package passed install/uninstall in the chocolatey test environment.
- [x] The changes only affect a single package (not including meta package).

### CODE REVIEW

File: automatic/kitty/tools/chocolateyInstall.ps1
Comment (majkinetor) [inline automatic/kitty/tools/chocolateyInstall.ps1:17]: Please remove those 2 comments, its obvious what commands do

File: automatic/kitty/tools/chocolateyInstall.ps1
Comment (majkinetor) [inline automatic/kitty/tools/chocolateyInstall.ps1:28]: remove comment

File: automatic/kitty/tools/helpers.ps1
Comment (majkinetor) [inline automatic/kitty/tools/helpers.ps1:5]: Reverse this logic please, so we don't have rest of the function idented.

File: automatic/kitty/tools/helpers.ps1
Comment (majkinetor) [inline automatic/kitty/tools/helpers.ps1:10]: Fixed sleep is automation is generally antipattern and should almost never be used.

File: automatic/kitty/tools/helpers.ps1
Comment (majkinetor) [inline automatic/kitty/tools/helpers.ps1:14]: Or you could create small inline function to be DRY.

File: automatic/kitty/update.ps1
Comment (majkinetor) [inline automatic/kitty/update.ps1:6]: Please put in this function what went into au_BeforeUpdate as per usual. Remove table from VERIFICATION.txt and use standard form.

File: automatic/kitty/update.ps1
Comment (majkinetor) [inline automatic/kitty/update.ps1:40]: Empty line after almost each line ?

File: automatic/kitty/update.ps1
Comment (majkinetor) [inline automatic/kitty/update.ps1:37]: Use Invoke-RestMethod

File: automatic/kitty/update.ps1
Comment (majkinetor) [inline automatic/kitty/update.ps1:47]: Please use standard updater.ps1 this looks convoluted.

**Review by majkinetor — CHANGES_REQUESTED** [review]

**Comment by majkinetor** [comment]
@yelkarama, thx for your effort, but some changes are required. 

You might want to take a look at some other updater and how it is setup, for example [nircmd](https://github.com/chocolatey-community/chocolatey-packages/blob/master/automatic/nomacs/update.ps1).

Several style problems should be corrected too.

File: automatic/kitty/tools/chocolateyInstall.ps1
Comment (yelkarama) [reply automatic/kitty/tools/chocolateyInstall.ps1]: Comments removed

File: automatic/kitty/tools/helpers.ps1
Comment (yelkarama) [reply automatic/kitty/tools/helpers.ps1:5]: Function simplified

File: automatic/kitty/tools/helpers.ps1
Comment (yelkarama) [reply automatic/kitty/tools/helpers.ps1]: Sleep removed

File: automatic/kitty/tools/helpers.ps1
Comment (yelkarama) [reply automatic/kitty/tools/helpers.ps1]: Function simplified

File: automatic/kitty/update.ps1
Comment (yelkarama) [reply automatic/kitty/update.ps1]: Removed empty lines

**Comment by RedBaron2** [comment]
@yelkarama 
The  update file will update all support files for the package via AU (ex: JSON and nuspec)

**Comment by yelkarama** [comment]
@RedBaron2 Any further actions required? 
Thanks

**Comment by RedBaron2** [comment]
@yelkarama 
Try to limit your changes to only the `update.ps1` file. All files should be only updated from AU instructions based in that update file. Test the update with AU inside the chocolatey test environment.

This is what makes for automatic updates of the many packages in this and other repositories.
Thanks for taking the initiative in this package.

**Comment by yelkarama** [comment]
@RedBaron2 The other changes are required. 

1. **chocolateyInstall.ps1** modified to handle new installation method (unzip the asset from github, instead of just downloading the individual binaries)
2. **helpers.ps1** added to provide a function called from chocolateyInstall.ps1 to stop runnion apps before the update
3. **VERIFICATION.txt** statndardized
4. **VERIFICATION_source.txt** not needed anymore
5. **gensfc.exe** not needed anymore

File: automatic/kitty/update.ps1
Comment (majkinetor) [inline automatic/kitty/update.ps1:22]: There is a AU function for this and standard method. See for example

https://github.com/majkinetor/au-packages/blob/057c7f8ede36ff4ad20b19f5c7ad91847fe1e894/rundeck/update.ps1#L22

**Review by majkinetor — CHANGES_REQUESTED** [review]
Improve hash stuff and we are done.

File: automatic/kitty/update.ps1
Comment (yelkarama) [reply automatic/kitty/update.ps1]: Done.

**Review by majkinetor — APPROVED** [review]
Its good now. Thank you.

**Comment by majkinetor** [comment]
I can not merge this, seems choco team removed my authorization lol :)

**Comment by yelkarama** [comment]
@majkinetor Who has the keys to merge this PR?

**Comment by majkinetor** [comment]
@yelkarama I don't know, given that I was **silently** removed which I became aware now, despite being top contributor and the one that automated most of this stuff.  Unless this is some error on choco team part, this is really weird behavior.

File: automatic/kitty/tools/chocolateyInstall.ps1
Comment (AdmiringWorm) [inline automatic/kitty/tools/chocolateyInstall.ps1:8]: We are trying to move towards removing globbing patterns when it is not needed.
Please update to use a actual path to the zip archive, and have that path being replaced in `update.ps1`.

See the `cmake.install` package as an example:

https://github.com/chocolatey-community/chocolatey-packages/blob/master/automatic/cmake.install/update.ps1#L23

File: automatic/kitty/tools/chocolateyInstall.ps1
Comment (AdmiringWorm) [inline automatic/kitty/tools/chocolateyInstall.ps1:17]: These calls to `Remove-Item` can be combined into one call.

```suggestion
Remove-Item -force "$toolsPath\*.zip", "$toolsPath\kitty-beta.exe", "$toolsPath\kitty.exe" -ea 0
```

**Review by AdmiringWorm — CHANGES_REQUESTED** [review]

**Comment by AdmiringWorm** [comment]
@majkinetor I just checked, and you have not been removed from the repository.

However, there have been changes that require approval of PRs, and all unresolved conversations to be resolved.
This, in turn, denies pushing directly to the repository unless the commit being made hasn't been reviewed.

I am aware this prevents creating fix versions temporarily, there are however plans to work around this, but not gotten that far just yet.

File: automatic/kitty/tools/chocolateyInstall.ps1
Comment (AdmiringWorm) [reply automatic/kitty/tools/chocolateyInstall.ps1:17]: Your point about the comment is fair, and I have updated my suggestion to keep it included.

However, having multiple remove-item calls when it is not needed is unnecessary, no matter how you look at it. So that suggestion is still relevant.

Comment (majkinetor) [reply automatic/kitty/tools/chocolateyInstall.ps1:17]: > when it is not needed is unnecessary

Its necessary, however you frame it, because semantics matter. By that logic, we can keep entire script in the same line because multiline code is not necessary, no matter how you look at it.

**Comment by AdmiringWorm** [comment]
> I did resolve all, and accepted changes via review, so your claims are not true.
> 
> image
> 
> image

If that second screenshot you have is from yesterday, then it is possible that there is a bug on GitHub or there are some requirements I got wrong.
Currently, no one can merge the PR, though, since there are conversations open regarding needed changes.

I'll take another look at the required checks after the changes have been added.

If it had been because you plainly didn't have permissions, the blocked message text would be different.

**Comment by majkinetor** [comment]
> If that second screenshot you have is from yesterday

It clearly is from yesterday, as shown on the screenshot and in GH itself.

>  then it is possible that there is a bug on GitHub or there are some requirements I got wrong

OK. 
Lets hope that it will be fixed soon then.

File: automatic/kitty/tools/chocolateyInstall.ps1
Comment (yelkarama) [reply automatic/kitty/tools/chocolateyInstall.ps1]: Done.

File: automatic/kitty/tools/chocolateyInstall.ps1
Comment (AdmiringWorm) [inline automatic/kitty/tools/chocolateyInstall.ps1:8]: The call to `Get-Item` should not be here anymore. However, I'll take care of it when merging it in.

**Review by AdmiringWorm — APPROVED** [review]
LGTM

**Comment by AdmiringWorm** [comment]
@yelkarama your changes have been merged, thanks for your contribution 👍

**Comment by majkinetor** [comment]
Is it fixed now since you merged it or this is just some fantasy?

**Comment by yelkarama** [comment]
@AdmiringWorm Interesting! I was not able to reproduce it on that same computer. I tried the following steps many times. No errors! 
1. Kill all running processes 
`Get-Process kitty,klink,kageant,kscp,ksftp,kittygen,cygtermd,genpass,gensfc -ea 0 | Stop-Process -Force`
2. Remove all existing versions of kitty:
`cuninst kitty`
3 Install previous version:
`cinst kitty --version 0.74.4.13`
4. Run kitty.exe and kageant.exe
5. Upgrade kitty
`cup kitty`

**Comment by AdmiringWorm** [comment]
hmm, it could have been something else into play that caused the locking in that case.
However, I think it is worthwhile getting a `chocolateyBeforeModify.ps1` script added nonetheless.

Since you can't reproduce it, can you then instead open a Feature/Enhancement discussion instead (https://github.com/chocolatey-community/chocolatey-packages/issues/new/choose), and we can take it from there.

**Pull request merged**
